logstash-core/src/main/java/org/logstash/plugins/aliases/AliasPlugin.java

```diff
@@ -0,0 +1,61 @@
+package org.logstash.plugins.aliases;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A POJO class to map AliasRegistry.yml structure.
+ */
+public class AliasPlugin {
+
+    /**
+     * Name of the aliased plugin.
+     */
+    @Nonnull
+    @JsonProperty("alias_name")
+    private String aliasName;
+
+    /**
+     * The plugin name where aliased plugin maps to.
+     */
+    @Nonnull
+    @JsonProperty("maps_to")
+    private String mapsTo;
+
+    /**
+     * List of <K,V> entries to replace when transforming artifact to aliased plugin.
+     */
+    @Nullable
+    private List<Map<String, String>> replaces;
+
+    @Nonnull
+    public String getAliasName() {
+        return aliasName;
+    }
+
+    public void setAliasName(@Nonnull String aliasName) {
+        this.aliasName = aliasName;
+    }
+
+    @Nonnull
+    public String getMapsTo() {
+        return mapsTo;
+    }
+
+    public void setMapsTo(@Nonnull String mapsTo) {
+        this.mapsTo = mapsTo;
+    }
+
+    @Nullable
+    public List<Map<String, String>> getReplaces() {
```

Comment:
I expect this method to be used from other code that needs to find a replacement for a given doc string of an aliased plugin.
The client code that uses this, has to understand the exposed structure to make its search. Instead of exposing internal structure I would create a method which brings the inputs needed, for example the string to find the replacement and the alias name, and keep the logic close to the data structure.

Some like:
```java
public String findAliasedReplacement(String aliasName, String docString)
```

This exposes the intention/behavior and encapsulates the data.

Reply:
@mashhurs nice re-shaping of the YAML structure and also the introduction of `AliasDocumentReplace` class.
I don't know how the client code would use these API, but if instead of exposing data to the client and let the client does the substitutions, what do you think about to have the client providing the string instance to work ok and let the `AliasPlugin` (which own the data) to do the replacement, acting as a String processor?
For example with a method like:
```java
class AliasPlugin {
....
public String replacesAliasesMarks(String doc) {
  // iterate on the internal data and return a string with the substitutions applied
}
```

Reply:
I don't know how the client code would use these API, but if instead of exposing data to the client and let the client does the substitutions, what do you think about to have the client providing the string instance to work ok and let the `AliasPlugin` (which own the data) to do the replacement, acting as a String processor?
For example with a method like:
```java
class AliasPlugin {
....
public String replacesAliasesMarks(String doc) {
  // iterate on the internal data and return a string with the substitutions applied
}
```

Reply:
I have removed this `get replaces` API since we are not using in code bases. Let's discuss when we have use cases.
For now, using replaces in doc-tooling by retrieving GH YAML content.